packages/bpk-component-graphic-promo/src/BpkGraphicPromo.js

```diff
@@ -0,0 +1,195 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkText from 'bpk-component-text';
+import BpkButton from 'bpk-component-button';
+import BpkCard from 'bpk-component-card';
+
+import TEXT_ALIGN from './BpkGraphicPromoTextAlign';
+import TEXT_COLORS from './BpkGraphicPromoTextColors';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  className: ?string,
+  kicker: ?string,
+  headline: string,
+  strapline: ?string,
+  image: string,
+  sponsor: ?{
+    label: string,
+    logo: string,
+    altText: string,
+  },
+  ctaText: string,
+  onClick: () => void,
+  invertVertically: boolean,
+  textAlign: TEXT_ALIGN,
+  textColor: TEXT_COLORS,
+};
+
+const constructAriaLabel = ({
+  ctaText,
+  headline,
+  kicker,
+  sponsor,
+  strapline,
+}: Props) => {
+  const text = [];
+  const addText = (value) => value && text.push(value);
+
+  if (sponsor) {
+    addText(sponsor.label);
+    addText(sponsor.altText);
+  } else {
+    addText(kicker);
+  }
+  addText(headline);
+  addText(strapline);
+  addText(ctaText);
+
+  return text.join('. ');
```

Comment:
Nitpick: Seems overkill to have defined and used a function to build up some text.

```javascript
const sponsoredText = [sponsor.label, sponsor.altText];
const kickerText = [kicker];
const commonText = [headline, strapline, ctaText];

const text = sponsor ? [...sponsoredText, ...commonText] : [...kickerText, ...commonText];

return text.join(' .');
```

Reply:
Maybe `const kickerText` is also overkill 🤔

Reply:
I see you also have a truthy-guard 🤔 

In combination with the `sponsor` check, is only `kicker` the other potentially falsey value?

Reply:
```javascript
const sponsoredText = [sponsor.label, sponsor.altText]; 
const kickerText = [kicker]; 
const commonText = [headline, strapline, ctaText]; 

const text = sponsor ? [...sponsoredText, ...commonText] : [...kickerText, ...commonText]; return text.join(' .');

return text.filter((t) => Boolean(t)).join(' .');
// or
return text.filter(Boolean).join(' .');
```

Reply:
It is just so as not to have to worry about things being defined or not. 😅

I am not trying to be defensive, but I believe having the `null`-check encapsulated in a function makes it more readable. Performance-wise neither of the two would matter much, so I guess we are only arguing readability here. 🙂

Reply:
Regarding your latest comment (sorry, we were typing simultaneously, I guess), it is only the two values that you mentioned. I just put it on everything to be extra cautious if someone messes the props up for whatever reason. This is not something that should happen, but I thought it helps make the code more concise and can catch things like "a design without a headline, where we hack it together by passing an empty string", for instance. 🙂

Reply:
Unless you have really strong feelings on this one @jonlaw101, I will resolve it?

Reply:
🤷

Reply:
Cool, I'll resolved it then. 😅